src/SimpleChoiceEnum.php

```diff
@@ -22,7 +22,9 @@
  */
 abstract class SimpleChoiceEnum extends ReadableEnum
 {
+    /* @use AutoDiscoveredValuesTrait<T> */
```

Comment:
Perhaps I should have use `/**` instead? or `//` ? Let me know.

Reply:
These annotations should be on the class level, right where you put `@extends`. It should also be `@uses` (with a **s**).

Reply:
It should be `/**` for docblocks indeed, but php-cs-fixer will revert it to `/*` since docblocks aren't usually allowed here.
I didn't find any documentation on `@use`, but shouldn't it rather be in the class's phpdoc using these traits?

Reply:
It's undocumented. I only found this: https://github.com/phpstan/phpstan-src/commit/87669234363d89252583664ef71309d5d8d3c7c1#diff-a2f75fb97bd3f8a99a1c56718a7ac23a03b9e6533eb4c6d34c4730f6d8e3132fR108.

For the phpdoc emplacement, for me I find it easier to set the `@use` juste above the place it's used to ease the maintenance: removing is no brainer as well as adding a new trait.

Moreover that's what is done in the commit I linked above.

Reply:
And `@uses` does not work, at least with PHPStan.

Reply:
Just opened an issue to document it: https://github.com/phpstan/phpstan/issues/5464

Reply:
Thanks, that explains a lot!